builtin/logical/mysql/path_roles.go

```diff
@@ -34,6 +34,11 @@ func pathRoles(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "SQL string to create a user. See help for more info.",
 			},
+
+			"username_length": &framework.FieldSchema{
```

Comment:
Making this as a mount level option might not be a good idea, because not all backends return a `username` per-se.